src/main/kotlin/com/team4099/robot2020/subsystems/Vision.kt

```diff
@@ -0,0 +1,113 @@
+package com.team4099.robot2020.subsystems
+
+import com.team4099.lib.logging.HelixLogger
+import edu.wpi.first.networktables.NetworkTable
+import edu.wpi.first.networktables.NetworkTableInstance
+import com.team4099.lib.subsystem.Subsystem
+import com.team4099.robot2020.config.Constants
+import edu.wpi.first.networktables.NetworkTableEntry
+import edu.wpi.first.wpilibj.controller.PIDController
+import kotlin.math.abs
+import kotlin.math.sign
+import kotlin.math.tan
+
+object Vision : Subsystem {
+    enum class VisionState {
+        IDLE, AIMING
+    }
+
+    var state = VisionState.IDLE
+        set(value) {
+            if (value != field) {
+                pipeline = when(value) {
+                    VisionState.IDLE -> {
+                        steeringAdjust = 0.0
+                        distanceAdjust = 0.0
+                        Constants.Vision.DRIVER_PIPELINE_ID
+                    }
+                    VisionState.AIMING -> Constants.Vision.TARGETING_PIPELINE_ID
+                }
+                field = value
+            }
+        }
+    private var distance = 0.0
+    private var distanceError = 0.0
+    var steeringAdjust = 0.0
+    var distanceAdjust = 0.0
+    var onTarget = false
+
+    private val table: NetworkTable = NetworkTableInstance.getDefault().getTable("limelight")
+    private val tx get() = table.getEntry("tx").getDouble(0.0)
+    private val ty get() = table.getEntry("ty").getDouble(0.0)
+    private val tv get() = table.getEntry("tv").getDouble(0.0) // is this a valid target?
+    private val ta get() = table.getEntry("ta").getDouble(0.0) // target area
+
+    private val pipelineEntry: NetworkTableEntry = table.getEntry("pipeline")
+    private val turnController = PIDController(
```

Comment:
do we need to zero the integral accumulator anywhere? ig only matters if we actually had a kI but no reason not to make it work